### PR TITLE
Add README, Contributing Guidelines and conda environment

### PR DESCRIPTION
**Pull request by prasunanand**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+conda env create -f .conda/environment.yml
+conda activate uarray
+```
+
+This will create an environment named `uarray` which you can use for development.
+
+`unumpy` and all development dependencies can be installed via:
+
+```
+pip install -e ".[all]"
```
Comment (hameerabbasi) [inline CONTRIBUTING.md]: We don't have a `setup.py` yet.

Comment (hameerabbasi) [reply CONTRIBUTING.md]: Well, now we do, but no `[all]` extra.

Comment (prasunanand) [reply CONTRIBUTING.md]: Resolved @hameerabbasi

**Comment by hameerabbasi** [comment]
Thanks @prasunanand!

**Pull request merged**